### PR TITLE
create the config file in the edit if it does not exist

### PR DESCRIPTION
**Pull request by ThunderKey**

### CODE REVIEW

File: src/kitovu/cli.py [hunk 1]
```diff
@@ -99,6 +99,13 @@ def edit(config: typing.Optional[pathlib.Path] = None, editor: typing.Optional[s
 
     if config is None:
         config = settings.get_config_file_path()
+        if not config.exists():
+            if not config.parent.exists():
```
Comment (The-Compiler) [inline src/kitovu/cli.py]: Both `.mkdir()` and `.touch()` have an `exists_ok=True` parameter (it's even `True` by default for `.touch()`), so you don't need these checks with that.

**Review by The-Compiler — CHANGES_REQUESTED** [review]
Not sure if this is needed at all - I'd expect you can easily pass a path which doesn't exist to an editor?

File: src/kitovu/cli.py (hunk 1 above)
Comment (ThunderKey) [reply src/kitovu/cli.py]: Ah, `exist_ok=True` makes more sense.

For example vim is not able to create the file if the folder `~/.config/kitovu` doesn't exist. So we could only create the folder, but I guess there are editors which require the file to exist (I think of not-so-smart Windows editors :wink:)

**Comment by lroellin** [comment]
This being open-source and me being a very interested future user, I like to chime in with @The-Compiler. This is something an editor should handle, not kitovu!

I even think it would be bad to do so. Many times I've opened vi when wanting to open a file, but tab-completion didn't find an existing file yet (ambiguous) and I wouldn't like it if a file like that would be created ;)

This being an EPJ feel *very* free to ignore my comments, but I had to chime in there, since I think it would be a design fault...

**Review by The-Compiler — APPROVED** [review]

**Comment by The-Compiler** [comment]
Thanks! You're right, there are probably editors which don't handle this *cough* Windows *cough* :wink:
As it's only the default config file location (and not if a custom `--config` has been given), I think it's a good idea to do this.

**Pull request merged**